prow/github/types.go

```diff
@@ -710,3 +710,23 @@ type Milestone struct {
 	Title  string `json:"title"`
 	Number int    `json:"number"`
 }
+
+// RepositoryCommit represents a commit in a repo.
+// Note that it's wrapping a GitCommit, so author/committer information is in two places,
+// but contain different details about them: in RepositoryCommit "github details", in GitCommit - "git details".
+type RepositoryCommit struct {
+	SHA         *string    `json:"sha,omitempty"`
+	Commit      *GitCommit `json:"commit,omitempty"`
+	Author      *User      `json:"author,omitempty"`
+	Committer   *User      `json:"committer,omitempty"`
+	Parents     []Commit   `json:"parents,omitempty"`
+	HTMLURL     *string    `json:"html_url,omitempty"`
+	URL         *string    `json:"url,omitempty"`
+	CommentsURL *string    `json:"comments_url,omitempty"`
+}
+
+// GitCommit represents a GitHub commit.
+type GitCommit struct {
+	SHA     *string `json:"sha,omitempty"`
+	Message *string `json:"message,omitempty"`
+}
```

Comment:
I have taken these structures from the Google GitHub API client. Not sure on usages of pointers here or if it's required - I notice that other structures here use a mixture, so left it as-is for now.

Reply:
I wouldn't use pointers unless you have a need to distinguish an omitted field from an empty field. If you don't use pointers you can't segfault :)
There might be a good reason for using pointers for everything though?